// azure-cli/lib/commands/arm/network/localNetworkGateway.js
/*** Generated by streamline 0.10.17 (callbacks) - DO NOT EDIT ***/ var __rt=require('streamline/lib/callbacks/runtime').runtime(__filename, false),__func=__rt.__func,__cb=__rt.__cb,__catch=__rt.__catch,__tryCatch=__rt.__tryCatch; var __ = require("underscore");















var util = require("util");
var utils = require("../../../util/utils");
var $ = utils.getLocaleString;
var VNetUtil = require("./../../../util/vnet.util");
var resourceUtils = require("../resource/resourceUtils");
var tagUtils = require("../tag/tagUtils");

function LocalNetworkGateway(cli, networkResourceProviderClient) {
  this.networkResourceProviderClient = networkResourceProviderClient;
  this.vnetUtil = new VNetUtil();
  this.output = cli.output;
  this.interaction = cli.interaction;};


__.extend(LocalNetworkGateway.prototype, {
  create: function create__1(resourceGroupName, name, options, _) { var self, parameters, gateway, __this = this; var __frame = { name: "create__1", line: 32 }; return __func(_, this, arguments, create__1, 3, __frame, function __$create__1() { self = __this;

      self._validate(options);

      parameters = self._parse(name, options);

      return self.get(resourceGroupName, name, __cb(_, __frame, 6, 23, function ___(__0, __1) { gateway = __1;
        if (gateway) {
          return _(new Error(util.format($("A local network gateway with name \"%s\" already exists in the resource group \"%s\""), name, resourceGroupName))); } ;


        return self._createOrUpdate(resourceGroupName, name, parameters, true, __cb(_, __frame, 11, 9, function __$create__1() {
          return self.show(resourceGroupName, name, options, __cb(_, __frame, 12, 9, function __$create__1() { _(); }, true)); }, true)); }, true)); }); },


  set: function set__2(resourceGroupName, name, options, _) { var self, gateway, parameters, __this = this; var __frame = { name: "set__2", line: 47 }; return __func(_, this, arguments, set__2, 3, __frame, function __$set__2() { self = __this;

      self._validate(options);

      return self.get(resourceGroupName, name, __cb(_, __frame, 4, 23, function ___(__0, __1) { gateway = __1;
        if (!gateway) {
          return _(new Error(util.format($("A local network gateway with name \"%s\" not found in the resource group \"%s\""), name, resourceGroupName))); } ;


        parameters = self._parse(name, options, gateway);

        return self._createOrUpdate(resourceGroupName, name, parameters, false, __cb(_, __frame, 11, 9, function __$set__2() {
          return self.show(resourceGroupName, name, options, __cb(_, __frame, 12, 9, function __$set__2() { _(); }, true)); }, true)); }, true)); }); },


  list: function list__3(resourceGroupName, options, _) { var self, networks, progress, __this = this; var __frame = { name: "list__3", line: 62 }; return __func(_, this, arguments, list__3, 2, __frame, function __$list__3() { self = __this;

      networks = null;

      progress = self.interaction.progress($("Looking up local networks")); return (function ___(__then) { (function ___(_) { __tryCatch(_, function __$list__3() {

            return self.networkResourceProviderClient.localNetworkGateways.list(resourceGroupName, __cb(_, __frame, 6, 73, function ___(__0, __1) { networks = __1; _(null, null, true); }, true)); }); })(function ___(__e, __r, __cont) { (function ___(__then) { __tryCatch(_, function __$list__3() {

              progress.end(); __then(); }); })(function ___() { __tryCatch(_, function ___() { if (__cont) { __then(); } else { _(__e, __r); }; }); }); }); })(function ___() { __tryCatch(_, function __$list__3() {


          self.interaction.formatOutput(networks.localNetworkGateways, function(data) {
            if ((data.length === 0)) {
              self.output.warn($("No local network gateways found")); }
             else {
              self.output.table(data, function(row, gateway) {
                row.cell($("Name"), gateway.name);
                row.cell($("Location"), gateway.location);
                row.cell($("IP Address"), gateway.gatewayIpAddress);
                var addressPrefixes = gateway.localNetworkAddressSpace.addressPrefixes;
                var address = addressPrefixes[0];
                if ((addressPrefixes.length > 1)) { address += ", ..."; };
                row.cell($("Address prefixes"), address); }); } ; }); _(); }); }); }); },





  show: function show__4(resourceGroupName, name, options, _) { var self, gateway, __this = this; var __frame = { name: "show__4", line: 90 }; return __func(_, this, arguments, show__4, 3, __frame, function __$show__4() { self = __this;

      return self.get(resourceGroupName, name, __cb(_, __frame, 2, 23, function ___(__0, __1) { gateway = __1;

        self.interaction.formatOutput(gateway, function(gateway) {
          if ((gateway === null)) {
            self.output.warn(util.format($("A local network gateway with name \"%s\" not found in the resource group \"%s\""), name, resourceGroupName)); }
           else {
            var resourceInfo = resourceUtils.getResourceInformation(gateway.id);
            self.output.nameValue($("Id"), gateway.id);
            self.output.nameValue($("Name"), gateway.name);
            self.output.nameValue($("Type"), resourceInfo.resourceType);
            self.output.nameValue($("Location"), gateway.location);
            self.output.nameValue($("Provisioning state"), gateway.provisioningState);
            self.output.nameValue($("Tags"), tagUtils.getTagsInfo(gateway.tags));
            self.output.nameValue($("IP Address"), gateway.gatewayIpAddress);
            self.output.header("Address prefixes");
            gateway.localNetworkAddressSpace.addressPrefixes.forEach(function(address) {
              self.output.listItem(address, 2); }); } ; }); _(); }, true)); }); },





  delete: function delete__5(resourceGroupName, name, options, _) { var self, gateway, progress, __this = this; var __frame = { name: "delete__5", line: 114 }; return __func(_, this, arguments, delete__5, 3, __frame, function __$delete__5() { self = __this;

      return self.get(resourceGroupName, name, __cb(_, __frame, 2, 23, function ___(__0, __2) { gateway = __2;

        if (!gateway) {
          return _(new Error(util.format($("A local network gateway with name \"%s\" not found in the resource group \"%s\""), name, resourceGroupName))); } ; return (function __$delete__5(_) {


          var __1 = !options.quiet; if (!__1) { return _(null, __1); } ; return self.interaction.confirm(util.format($("Delete local network gateway \"%s\"? [y/n] "), name), __cb(_, __frame, 8, 44, function ___(__0, __3) { var __2 = !__3; return _(null, __2); }, true)); })(__cb(_, __frame, -113, 17, function ___(__0, __3) { return (function __$delete__5(__then) { if (__3) { return _(null); } else { __then(); } ; })(function __$delete__5() {



            progress = self.interaction.progress(util.format($("Deleting local network gateway \"%s\""), name)); return (function ___(__then) { (function ___(_) { __tryCatch(_, function __$delete__5() {

                  return self.networkResourceProviderClient.localNetworkGateways.deleteMethod(resourceGroupName, name, __cb(_, __frame, 14, 62, function __$delete__5() { _(null, null, true); }, true)); }); })(function ___(__e, __r, __cont) { (function ___(__then) { __tryCatch(_, function __$delete__5() {

                    progress.end(); __then(); }); })(function ___() { __tryCatch(_, function ___() { if (__cont) { __then(); } else { _(__e, __r); }; }); }); }); })(function ___() { __tryCatch(_, function __$delete__5() { _(); }); }); }); }, true)); }, true)); }); },



  get: function get__6(resourceGroupName, name, _) { var self, progress, gateway, __this = this; var __frame = { name: "get__6", line: 134 }; return __func(_, this, arguments, get__6, 2, __frame, function __$get__6() { self = __this;

      progress = self.interaction.progress(util.format($("Looking up local network gateway \"%s\""), name)); return (function ___(__then) { (function ___(_) { __tryCatch(_, function __$get__6() { return (function ___(__then) { (function ___(_) { __tryCatch(_, function __$get__6() {


                  return self.networkResourceProviderClient.localNetworkGateways.get(resourceGroupName, name, __cb(_, __frame, 5, 76, function ___(__0, __1) { gateway = __1;
                    return _(null, gateway.localNetworkGateway); }, true)); }); })(function ___(e, __result) { __catch(function __$get__6() { if (e) {

                    if ((e.statusCode === 404)) {
                      return _(null, null); } ;

                    return _(e); } else { _(null, __result); } ; }, _); }); })(function ___() { __tryCatch(_, function __$get__6() { _(null, null, true); }); }); }); })(function ___(__e, __r, __cont) { (function ___(__then) { __tryCatch(_, function __$get__6() {

              progress.end(); __then(); }); })(function ___() { __tryCatch(_, function ___() { if (__cont) { __then(); } else { _(__e, __r); }; }); }); }); })(function ___() { __tryCatch(_, function __$get__6() { _(); }); }); }); },



  _createOrUpdate: function _createOrUpdate__7(resourceGroupName, name, parameters, isCreating, _) { var self, action, progress, __this = this; var __frame = { name: "_createOrUpdate__7", line: 151 }; return __func(_, this, arguments, _createOrUpdate__7, 4, __frame, function __$_createOrUpdate__7() { self = __this;

      action = (isCreating ? "Creating" : "Updating");
      progress = self.interaction.progress(util.format($("%s local network gateway \"%s\""), action, name)); return (function ___(__then) { (function ___(_) { __tryCatch(_, function __$_createOrUpdate__7() {


            return self.networkResourceProviderClient.localNetworkGateways.createOrUpdate(resourceGroupName, name, parameters, __cb(_, __frame, 6, 62, function __$_createOrUpdate__7() { _(null, null, true); }, true)); }); })(function ___(__e, __r, __cont) { (function ___(__then) { __tryCatch(_, function __$_createOrUpdate__7() {

              progress.end(); __then(); }); })(function ___() { __tryCatch(_, function ___() { if (__cont) { __then(); } else { _(__e, __r); }; }); }); }); })(function ___() { __tryCatch(_, function __$_createOrUpdate__7() { _(); }); }); }); },



  _validate: function(options) {
    var self = this;

    if (options.ipAddress) {
      var ipValidation = self.vnetUtil.parseIPv4(options.ipAddress, "--ip-address");
      if (ipValidation.error) {
        throw new Error(ipValidation.error); } ; } ;



    if (options.addressSpace) {
      options.addressSpace.split(",").forEach(function(addressPrefix) {
        var cidrValidation = self.vnetUtil.parseIPv4Cidr(addressPrefix, "--address-space");
        if (cidrValidation.error) {
          throw new Error(cidrValidation.error); } ; }); } ; },





  _parse: function(name, options, gateway) {
    var parameters = {
      localNetworkAddressSpace: {
        addressPrefixes: [] },

      gatewayIpAddress: "",
      location: "",
      tags: { } };


    if (gateway) { parameters = gateway; };

    if (options.addressSpace) {
      options.addressSpace.split(",").forEach(function(addressPrefix) {
        parameters.localNetworkAddressSpace.addressPrefixes.push(addressPrefix); }); } ;



    if (options.ipAddress) {
      parameters.gatewayIpAddress = options.ipAddress; } ;


    if (options.location) {
      parameters.location = options.location; } ;


    if (options.tags) {
      var tags = tagUtils.buildTagsParameter(null, options);
      tagUtils.appendTags(parameters, tags); } ;


    if ((options.tags === false)) {
      gateway.tags = { }; } ;


    return parameters; }});



module.exports = LocalNetworkGateway;
